Chatsagnik

chatsagnik opened the pull request:
Undo Functionality achieved!
Work To do -> Redo & RegEx
From Sagnik and Soumyarup
B)

Commit:
Changed editor.c

Commit:
Cleaned up editor.c and made the loading faster

Commit:
added new APIs for displaying the LL

chatsagnik committed:
started on the Undo Redo functionalities

chatsagnik committed:
revised the URnode functionality

fleshed out the URnode function

chatsagnik committed:
added undo and redo stack operation functions

jayantbh committed:
chatsagnik: Errors resolved. Please check committers notes in editor.c.

chatsagnik committed:
Update editor.c

Resolved compilation errors.

chatsagnik committed:
Update editor.c

Undo Functionality achieved!
Work To do -> Redo & RegEx
From Sagnik and Soumyarup
B)

jayantbh commented on editor.c:
```diff
@@ -1,175 +1,298 @@
+	*/
+	char operation1 = undoRedoNode->operation;
+	switch(operation1){
+		case 'I':
```
We could use a full string for operations right? Like, `INSERT`, `DELETE`, `MOVE`, etc.
I suggest being more verbose here. What do you think?

jayantbh commented on editor.c:
```diff
@@ -1,175 +1,298 @@
+	//Load the whole file into a LinkedList and returns the tail pointer
+	struct node* currNode = loadFileToList(argv[1]);
+	insertCharAfter(currNode,'H');currNode = moveCursor(currNode,1);//currNode=currNode->next;
+	insertCharAfter(currNode,'I');currNode = moveCursor(currNode,1);//currNode=currNode->next;
```
Same, verbosity with operations key. Your opinion?

jayantbh commented:
@chatsagnik resolve conflicts please.

jayantbh committed:
Merge remote-tracking branch 'origin/chatsagnik' into chatsagnik

# Conflicts:
#	source/editor.c

jayantbh commented:
@chatsagnik check if the current revision looks like that you intended.

chatsagnik committed:
updated editor.c

fixed all problems with insert and delete
working on integrating undo and redo with move

chatsagnik committed:
Delete editable.c

chatsagnik committed:
Update editor.c

modularised the code

chatsagnik committed:
Create listfunc.h

chatsagnik committed:
Update editor.c

chatsagnik committed:
Update listfunc.h

chatsagnik committed:
Update editor.c

chatsagnik committed:
Update listfunc.h
